python/pyspark/sql/types.py

```diff
@@ -2197,8 +2197,10 @@ def verify_nullability(obj: Any) -> bool:
                 return True
             else:
                 raise PySparkValueError(
-                    error_class="CANNOT_BE_NONE",
-                    message_parameters={"arg_name": "obj"},
+                    error_class="FIELD_NOT_NULLABLE",
+                    message_parameters={
+                        "field_name": name if name is not None else "",
```

Comment:
Seems like the error message would look weird if this is an empty string.

Reply:
Separated into two error class `FIELD_NOT_NULLABLE` and `FIELD_NOT_NULLABLE_WITN_NAME`. Please let me know if any other suggestions!